### PR TITLE
fix(api): handle reasoning models in suggested questions generation

### PR DESCRIPTION
**Pull request by plind-junior**
## Summary

Fixes #34475

The "Suggested Questions After Answer" feature silently fails when the workspace's System Reasoning Model is a reasoning model (e.g. DeepSeek-R1, o1). The root cause is twofold:

1. **Token budget exhaustion**: Reasoning models consume completion tokens for their internal `<think>` process. The hardcoded `max_tokens=256` is entirely consumed by thinking, leaving no tokens for the actual JSON output.

2. **Unparsed reasoning tags**: Some providers include `<think>...</think>` blocks in the text content rather than in the separate `reasoning_content` field. The output parser did not strip these tags, causing the JSON extraction regex to fail.

### Changes

- **`api/core/llm_generator/prompts.py`**: Increase default `SUGGESTED_QUESTIONS_MAX_TOKENS` from `256` to `512`. The value remains overridable via the `SUGGESTED_QUESTIONS_MAX_TOKENS` environment variable.
- **`api/core/llm_generator/output_parser/suggested_questions_after_answer.py`**: Strip `<think>...</think>` blocks from the LLM response before extracting the JSON array. Note: `structured_output.py` handles a similar problem differently — it relies on `json_repair.loads()` to implicitly skip non-JSON prefixes like `<think>` blocks. The explicit regex approach here is more robust for this use case since the parser needs to extract a JSON array from free-form text rather than parsing a single JSON object.
- **`api/tests/unit_tests/core/llm_generator/test_llm_generator.py`**: Add test cases covering `<think>`-prefixed output, large think blocks (>= 2000 tokens), unclosed/truncated think tags, empty think tags, and nested think tags.

## Test plan

- [x] All existing unit tests pass
- [x] New tests cover `<think>`-prefixed output from reasoning models and edge cases
- [x] Manual verification: set a reasoning model (e.g. DeepSeek-R1) as System Reasoning Model, send a message in a Chatflow with "Suggested Questions After Answer" enabled, and verify follow-up questions appear

### CODE REVIEW

File: api/core/llm_generator/output_parser/suggested_questions_after_answer.py [hunk 1]
```diff
@@ -7,12 +7,16 @@
...
+_THINK_TAG_RE = re.compile(r"<think>.*?</think>", re.DOTALL)
+
 
 class SuggestedQuestionsAfterAnswerOutputParser:
     def get_format_instructions(self) -> str:
         return SUGGESTED_QUESTIONS_AFTER_ANSWER_INSTRUCTION_PROMPT
 
     def parse(self, text: str) -> Sequence[str]:
+        # Strip reasoning model <think> blocks before parsing
+        text = _THINK_TAG_RE.sub("", text)
```
[the remark this answers is not in the document]
Comment (plind-junior) [reply api/core/llm_generator/output_parser/suggested_questions_after_answer.py:19]: structured_output.py doesn't actually use a regex to strip think tags — it relies on json_repair.loads() to handle the malformed JSON. The two approaches are fundamentally different. Extracting a shared helper would mean refactoring structured_output.py too, which is out of scope for this bug fix. The duplication is minimal (one regex + one .sub() call).

[the remark this answers is not in the document]
Comment (plind-junior) [reply api/core/llm_generator/output_parser/suggested_questions_after_answer.py:19]: same as above

**Comment by plind-junior** [comment]
@asukaminato0721 Would you check my PR?

**Comment by plind-junior** [comment]
@asukaminato0721 @fatelei @laipz8200   Would you check this PR?

File: api/tests/unit_tests/core/llm_generator/test_llm_generator.py [hunk 2]
```diff
@@ -97,6 +97,19 @@ def test_generate_suggested_questions_after_answer_success(self, mock_model_inst
         assert len(questions) == 2
         assert questions[0] == "Question 1?"
 
+    def test_generate_suggested_questions_after_answer_with_think_tags(self, mock_model_instance):
+        mock_response = MagicMock()
+        mock_response.message.get_text_content.return_value = (
+            "<think>\nThe user asked about Python.\nI should suggest related questions.\n</think>\n"
+            '["What are Python decorators?", "How does async work in Python?"]'
+        )
+        mock_model_instance.invoke_llm.return_value = mock_response
```
[the remark this answers is not in the document]
Comment (crazywoola) [reply api/tests/unit_tests/core/llm_generator/test_llm_generator.py:106]: It only covers the happy path of this scenario. So you should add more tests regarding this feature.

- the thinking  tokens >= 2000 tokens
- the thinking process stop

**Review by crazywoola — CHANGES_REQUESTED** [review]
See comments

**Comment by crazywoola** [comment]
@dosu review this PR

File: api/tests/unit_tests/core/llm_generator/test_llm_generator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (plind-junior) [reply api/tests/unit_tests/core/llm_generator/test_llm_generator.py:106]: Added both cases in the latest push:

Large think block (>= 2000 tokens): test_generate_suggested_questions_after_answer_with_large_think_block — uses ~200 repeated sentences (~2000+ tokens) inside a closed <think> block. Regex strips it and questions parse correctly.

Thinking process stops (unclosed <think>): test_generate_suggested_questions_after_answer_with_unclosed_think_tag — no closing </think>, simulating truncated reasoning output. The regex doesn't match (non-greedy requires closing tag), so the text passes through as-is and the JSON array is still extracted by the \[.*?\] pattern.

Also added tests for empty <think></think> and nested <think> tags for completeness.

**Comment by plind-junior** [comment]
@crazywoola @laipz8200 Reviewed the comment and updated

**Comment by plind-junior** [comment]
@asukaminato0721 Sorry for the interrupt during the weekend but this PR has been opened for a week.
Do you have a chance to review this PR? ty

**Comment by plind-junior** [comment]
@crazywoola Would you review the update?

**Comment by plind-junior** [comment]
@laipz8200 @crazywoola It's been ready for review for the last 2 weeks. Would you reivew it?

**Comment by Mister-Hope** [comment]
I would like to see this merged. I meet similar problems when trying to integrate deepseek-v4-flash with the suggestions, but gemini 3.5 flash lite works fine.

This is kind of annoying.
